fastai/datasets.py

```diff
@@ -64,7 +64,11 @@ def url2name(url): return url.split('/')[-1]
 def _url2path(url, data=True):
     name = url2name(url)
     return datapath4file(name) if data else modelpath4file(name)
-def _url2tgz(url): return datapath4file(f'{url2name(url)}.tgz')
+def _url2tgz(url):
```

Comment:
Generally we prefer to avoid using more vertical space than necessary. E.g. here this can simply something like:
```
ext = "" if url.endswith((".tar.gz", ".tgz")) else ".tgz"') 
return datapath4file(f'{url2name(url)}{ext}')
```

Reply:
BTW the style guide is here if you're interested: https://docs-dev.fast.ai/style.html

Reply:
Thanks for your patient instruction.  I'm happy to comply to the style guide and will gladly do as you ask -- and/or feel free to simply add that code yourself and delete my PR.  

I would like to learn how to write the test as well. I don't see anything comparable for _url2tgz() currently in [fastai/tests/test_datasets.py](https://github.com/fastai/fastai/blob/master/tests/test_datasets.py), and having now read the ["Writing tests" section of the testing guide](https://docs-dev.fast.ai/test.html#writing-tests)...I don't see how it applies.  Is there a different guide I should be reading for this?   Thanks. 

...and now that I modified the code and pushed it, I see your [git guide](https://docs-dev.fast.ai/git.html#step-2-create-a-branch), and have already pushed to the master branch which I was not supposed to do.  I will follow the instructions to switch to a new branch...